### PR TITLE
fix[lsp-fsharp]: change permissions of fsac files

### LINKED ISSUE #3069: [F#] FSAC with strict permissions has execution denied
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [ ] You may also try reproduce the issue using clean environment using the following command
```bash
emacs -q -l lsp-start-plain.el
```
where `lsp-start-plain.el` can be downloaded [here](https://github.com/emacs-lsp/lsp-mode/blob/master/scripts/lsp-start-plain.el).

## Problem
- On Fsharp, fsautocomplete might be downloaded without permissions to execute the dotnet dlls.
- Refer to https://github.com/fsharp/FsAutoComplete/issues/715

## Fix
Under such circumstances, simply altering the permission of the files in the cache folder might solve the issue

## Screenshots
Screenshot from 2021-08-28 19-28-47
Screenshot from 2021-08-28 19-29-28

## Tested on
- NixOS (non-root user)
- Debian 11 docker image (non-root user)

### Steps to reproduce

- mkdir Test && cd Test
- dotnet new console -lang F#
- open `Program.fs` with emacs and start lsp with fsac (default option) 

### Expected behavior

- Lsp-mode should simply be able to execute the dll file and its json dependencies

### Which Language Server did you use?

lsp-fsharp

### OS

Linux

### Error callstack

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by MMagueta**
Fixes #3069

### CODE REVIEW

**Review by jcs090218 — APPROVED** [review]

File: clients/lsp-fsharp.el [hunk 1]
```diff
@@ -205,6 +205,16 @@ current runtime."
                   "fsautocomplete.netcore.zip"
                 "fsautocomplete.zip")))
 
+(defun lsp-fsharp--change-permissions (install-dir-full)
+  (unless (eq system-type 'windows-nt) ; Windows does not have chmod
+    (progn
```
Comment (yyoncho) [inline clients/lsp-fsharp.el:210]: nit: no need from progn.

**Review by yyoncho — APPROVED** [review]

File: clients/lsp-fsharp.el [hunk 2]
```diff
@@ -205,6 +205,16 @@ current runtime."
...
+      (message "Altering permissions")
```
Comment (yyoncho) [inline clients/lsp-fsharp.el:211]: typically, we use lsp--info instead of message.

File: clients/lsp-fsharp.el (hunk 1 above)
Comment (MMagueta) [reply clients/lsp-fsharp.el]: good catch

File: clients/lsp-fsharp.el (hunk 2 above)
Comment (MMagueta) [reply clients/lsp-fsharp.el]: Updated. Consistency IS important :)

**Comment by yyoncho** [comment]
Thank you for contributing to `lsp-mode`!

**Comment by MMagueta** [comment]
> Thank you for contributing to `lsp-mode`!

Thank you for maintaining this!

**Pull request merged**